sdk/storage/Azure.Storage.Files.DataLake/src/PathClient.cs

```diff
@@ -0,0 +1,1462 @@
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// every request.
+        /// </param>
+        public PathClient(Uri pathUri, StorageSharedKeyCredential credential, DataLakeClientOptions options = default)
+            : this(pathUri, credential.AsPolicy(), options)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="pathUri">
+        /// A <see cref="Uri"/> referencing the path that includes the
+        /// name of the account, the name of the file system, and the path to
+        /// the resource.
+        /// </param>
+        /// <param name="authentication">
+        /// An optional authentication policy used to sign requests.
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// every request.
+        /// </param>
+        internal PathClient(Uri pathUri, HttpPipelinePolicy authentication, DataLakeClientOptions options)
+        {
+            _uri = pathUri;
+            _blobUri = GetBlobUri(pathUri);
+            _dfsUri = GetDfsUri(pathUri);
+            _pipeline = (options ?? new DataLakeClientOptions()).Build(authentication);
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _blockBlobClient = new BlockBlobClient(_blobUri, _pipeline, _clientDiagnostics, null);
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="pathUri">
+        /// A <see cref="Uri"/> referencing the directory that includes the
+        /// name of the account, the name of the file system, and the path to the
+        /// resource.
+        /// </param>
+        /// <param name="pipeline">
+        /// The transport pipeline used to send every request.
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// </param>
+        internal PathClient(Uri pathUri, HttpPipeline pipeline, DataLakeClientOptions options = default)
+        {
+            _uri = pathUri;
+            _blobUri = GetBlobUri(pathUri);
+            _dfsUri = GetDfsUri(pathUri);
+            _pipeline = pipeline;
+            _clientDiagnostics = new ClientDiagnostics(options ?? new DataLakeClientOptions());
+            _blockBlobClient = new BlockBlobClient(_blobUri, pipeline, _clientDiagnostics, null);
+        }
+        #endregion
+
+        /// <summary>
+        /// Gets the blob Uri.
+        /// </summary>
+        private static Uri GetBlobUri(Uri uri)
```

Comment:
We've got these a couple of places.  Let's have a shared implementation on DataLakeUriBuilder.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8386